### PR TITLE
Feature/model 2

### PR DESCRIPTION
**Pull request by viethuynh317**

### CODE REVIEW

File: rpi-dp/leaveprob-dttor/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+lib/
+dataset/
+test/
+*.model
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/.gitignore:4]: Add a newline

File: rpi-dp/leaveprob-dttor/lenet.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        # FC => RELU
+        model.add(Flatten())
+        model.add(Dense(500))
+        model.add(Activation('relu'))
+
+        # softmax clfier
+        model.add(Dense(classes))
+        model.add(Activation('softmax'))
+
+        return model
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/lenet.py:37]: Add a newline

File: rpi-dp/leaveprob-dttor/plantcondition_recognition.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    prediction = model.predict(image)[0]
+
+    labels = ('problematicplant', 'goodplant')
+    pred_dict = list(zip(labels, tuple(prediction)))
+
+    (label, proba) = max(pred_dict, key=lambda item: item[1])
+    return (label, proba)
+
+
+
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/plantcondition_recognition.py:30]: Remove redundant blank lines.

File: rpi-dp/leaveprob-dttor/network_testing.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# network testing
+# based on this article on pyimagesearch
+# https://www.pyimagesearch.com/2017/12/11/image-classification-with-keras-and-deep-learning/
+
+# To test the network from terminal, make sure you are at rpi-dp/planttp-clfier/, if not cd there and run:
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/network_testing.py:5]: Change `rpi-dp/planttp-clfier/` to `rpi-dp/leaveprob-dttor/`

File: rpi-dp/leaveprob-dttor/network_testing.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+output = imutils.resize(original, width=400)
+cv2.putText(
+    output, label_on_img, (10, 25),
+    cv2.FONT_HERSHEY_SIMPLEX,
+    0.6, (255, 25, 0), 2
+)
+
+print(list(pred_dict))
+cv2.imshow('Output image', output)
+cv2.waitKey(0)
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/network_testing.py:57]: Add a blank line at the end of this file.

File: rpi-dp/planttp-clfier/planttp_recognition.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    prediction = model.predict(image)[0]
+
+    labels = ('succulent', 'palmplant', 'flower', 'foliageplant')
+    pred_dict = list(zip(labels, tuple(prediction)))
+
+    (label, proba) = max(pred_dict, key=lambda item: item[1])
+    return (label, proba)
+
+
+
```
Comment (khoaDLuu) [inline rpi-dp/planttp-clfier/planttp_recognition.py:30]: Remove 2 redundant blank lines.

File: rpi-dp/leaveprob-dttor/network_training.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+N = EPOCHS
+plt.plot(np.arange(0, N), H.history["loss"], label="train_loss")
+plt.plot(np.arange(0, N), H.history["val_loss"], label="val_loss")
+plt.plot(np.arange(0, N), H.history["accuracy"], label="train_acc")
+plt.plot(np.arange(0, N), H.history["val_accuracy"], label="val_acc")
+plt.title("Training Loss and Accuracy on Plant condition")
+plt.xlabel("Epoch #")
+plt.ylabel("Loss/Accuracy")
+plt.legend(loc="lower left")
+plt.savefig(args["plot"])
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/network_training.py:130]: Add a blank line.

File: rpi-dp/leaveprob-dttor/network_training.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
+# network training
+
+# To train the network from terminal, make sure you are at rpi-dp/planttp-clfier/, if not cd there and run:
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/network_training.py:3]: Change `rpi-dp/planttp-clfier/` to `rpi-dp/leaveprob-dttor/`.

File: rpi-dp/leaveprob-dttor/lenet.py (hunk 2 above)
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/lenet.py:37]: Add a blank line here.

File: rpi-dp/leaveprob-dttor/.gitignore (hunk 1 above)
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/.gitignore:4]: Add a blank line here.

File: rpi-dp/leaveprob-dttor/plantcondition_recognition.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+from keras.preprocessing.image import img_to_array
+from keras.models import load_model
+import numpy as np
+import imutils
+import cv2
+
+import time
+import picamera
+
+def planttp_rec(image, model_path):
```
Comment (khoaDLuu) [inline rpi-dp/leaveprob-dttor/plantcondition_recognition.py:12]: Rename this function to `plantcond_rec`.

File: rpi-dp/leaveprob-dttor/.gitignore (hunk 1 above)
Comment (viethuynh317) [reply rpi-dp/leaveprob-dttor/.gitignore:4]: ok

File: rpi-dp/leaveprob-dttor/plantcondition_recognition.py (hunk 9 above)
Comment (viethuynh317) [reply rpi-dp/leaveprob-dttor/plantcondition_recognition.py:12]: ok

File: rpi-dp/planttp-clfier/planttp_recognition.py (hunk 6 above)
Comment (viethuynh317) [reply rpi-dp/planttp-clfier/planttp_recognition.py:30]: ok

File: rpi-dp/leaveprob-dttor/network_testing.py (hunk 5 above)
Comment (viethuynh317) [reply rpi-dp/leaveprob-dttor/network_testing.py:57]: ok

File: rpi-dp/leaveprob-dttor/network_testing.py (hunk 4 above)
Comment (viethuynh317) [reply rpi-dp/leaveprob-dttor/network_testing.py:5]: ok

File: rpi-dp/leaveprob-dttor/plantcondition_recognition.py (hunk 3 above)
Comment (viethuynh317) [reply rpi-dp/leaveprob-dttor/plantcondition_recognition.py:30]: ok

**Pull request merged**